app/src/main/kotlin/com/softteco/template/utils/HandleApiErrorExtension.kt

```diff
@@ -0,0 +1,29 @@
+package com.softteco.template.utils
+
+import com.softteco.template.R
+import com.softteco.template.data.base.error.ErrorEntity
+import com.softteco.template.data.base.error.Result
+import com.softteco.template.ui.components.SnackBarState
+import kotlinx.coroutines.flow.MutableStateFlow
+
+fun <T> handleApiError(
+    result: Result<T>,
+    errorStateFlow: MutableStateFlow<SnackBarState>
+) {
+    when (result) {
+        is Result.Error -> {
+            val textId = when (result.error) {
+                is ErrorEntity.AccessDenied -> R.string.error_example
```

Comment:
We want to avoid duplication in error handling, but in fact we just moved it from ViewModel. I think it's a good time to create meaningful messages for all cases

Reply:
Are these messages ok? 

ErrorEntity.AccessDenied:
"Sorry, but you don't have the necessary permissions to access this resource."

ErrorEntity.Network:
Please check your internet connection and try again."

ErrorEntity.NotFound:
"The requested resource could not be found. Please make sure you've entered the correct information."

ErrorEntity.ServiceUnavailable:
 "Our services are temporarily unavailable. We apologize for the inconvenience and are working to resolve the issue."

ErrorEntity.Unknown:
"An unexpected error occurred. We're investigating the issue, and we apologize for any inconvenience."

Reply:
They're generally good, but since we're using Snackbar to display errors, we may need to check if they fit on two lines, as the [Snackbar guidelines recommend](https://m3.material.io/components/snackbar/guidelines#1dec52e3-5641-4a7d-a708-554889466850)

Reply:
Not really understand for what should we check if our Snackbar  displays multi-line text? For SnackbarHost we don't specify `size` parameter, so the modifier will make the content fill the whole available size

Reply:
From the Material3 guidelines:
"On mobile, use up to two lines of text to communicate the snackbar message"

Reply:
if the message fits on more than two lines, what should we do?